jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/CorrectThreadCredentialsProvider.kt

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.util.ThrowableComputable
+import software.amazon.awssdk.auth.credentials.AwsCredentials
+import software.aws.toolkits.core.credentials.ToolkitCredentialsProvider
+import software.aws.toolkits.resources.message
+
+/**
+ * Offloads fetching credentials to a background task and a modal progress bar if the current thread is EDT
+ */
+internal class CorrectThreadCredentialsProvider(private val delegate: ToolkitCredentialsProvider) :
+    ToolkitCredentialsProvider() {
+
+    override val id: String
+        get() = delegate.id
+
+    override val displayName: String
+        get() = delegate.displayName
+
+    override fun resolveCredentials(): AwsCredentials {
+        return if (ApplicationManager.getApplication().isDispatchThread) {
+            ProgressManager.getInstance(). runProcessWithProgressSynchronously(
+                ThrowableComputable<AwsCredentials, Exception> {
+                    delegate.resolveCredentials()
+                },
+                message("credentials.retrieving"),
+                /* canBeCancelled */false,
```

Comment:
Why comments over named arguments?

Reply:
Cant use named arguments when function is defined in java